samples/bootloader/src/main.c

```diff
@@ -58,13 +59,19 @@ static void boot_from(const struct fw_info *fw_info)
 		return;
 	}
 
+	if (fw_info->version > get_monotonic_counter()) {
```

Comment:
Should we check if CONFIG_NUM_COUNTER_SLOTL == 0, here? If it is, then there is no need to include this code.

Reply:
CONFIG_NUM_COUNTER_SLOTS is an app config, and the value is reflected through the provision page, so it's not strictly known at compile time.